dartagnan/src/main/java/com/dat3m/dartagnan/wmm/Wmm.java

```diff
@@ -152,33 +155,35 @@ public BoolExpr encodeBase(Context ctx) {
             recursiveGroup.updateEncodeTupleSets();
         }
 
-        BoolExpr enc = ctx.mkTrue();
+        BooleanFormulaManager bmgr = ctx.getFormulaManager().getBooleanFormulaManager();
+		BooleanFormula enc = bmgr.makeTrue();
         for(String relName : baseRelations){
-            enc = ctx.mkAnd(enc, relationRepository.getRelation(relName).encode(ctx));
+            enc = bmgr.and(enc, relationRepository.getRelation(relName).encode(ctx));
         }
 
         return enc;
     }
 
     // Initalizes everything just like encodeBase but also encodes all
     // relations that are needed for the axioms (but does NOT encode the axioms themselves yet)
-    public BoolExpr encode(Context ctx) {
-        BoolExpr enc = encodeBase(ctx);
+    public BooleanFormula encode(SolverContext ctx) {
+    	BooleanFormula enc = encodeBase(ctx);
         for (Axiom ax : axioms) {
-            enc = ctx.mkAnd(enc, ax.getRelation().encode(ctx));
+            enc = ctx.getFormulaManager().getBooleanFormulaManager().and(enc, ax.getRelation().encode(ctx));
```

Comment:
Move ctx.getFormulaManager().getBooleanFormulaManager out of the loop (just like in the method "consistent")